Bottom: bc98e32c7ce9614150acb9f1e6dbefa1762c1634
Top:    43dda453f9f1db191fad4fbf43e9e1316d2aa76f
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 20:47:43 -0400

Refresh of mpc5200-ac97

---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 7253303..00e832f 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -357,24 +357,21 @@ reset:
 	}
 
 	psc_ac97_cold_reset(&ac97);
+	psc_ac97_warm_reset(&ac97);
 
 	/* first make sure it is low */
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) != 0) {
 		udelay(1);
-		if (timeout++ > 1000) {
-			psc_ac97_warm_reset(&ac97);
+		if (timeout++ > 1000)
 			goto reset;
-		}
 	}
 	/* then wait for the transition to high */
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) == 0) {
 		udelay(1);
-		if (timeout++ > 1000) {
+		if (timeout++ > 1000)
 			psc_ac97_warm_reset(&ac97);
-			goto reset;
-		}
 	}
 
 	out_8(&regs->command, MPC52xx_PSC_TX_ENABLE | MPC52xx_PSC_RX_ENABLE);
